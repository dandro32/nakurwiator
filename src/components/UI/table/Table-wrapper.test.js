import React from 'react';
import TableWrapper from './Table-wrapper';
import TestRenderer from 'react-test-renderer';
import { faDumbbell} from '@fortawesome/free-solid-svg-icons';


jest.mock('react-table', () => 'ReactTable');
jest.mock('./Table-action-cell', () => 'TableActionCell');

describe('TableActionCell component tests', () => {
  let json;
  const mockData = [1,2,3];
  const mockColumns = [{},{}];
  const deleteStubFunc = jest.fn();

  beforeEach(() => {
    const renderer = TestRenderer.create(
      <TableWrapper isLoaded={true} icon={faDumbbell} title='test' data={mockData} columns={mockColumns} deleteRow={deleteStubFunc}/>
    );
    json = renderer.toJSON();
  });

  it('should render properly component output', () => {
    expect(json).toMatchSnapshot();
  });



});