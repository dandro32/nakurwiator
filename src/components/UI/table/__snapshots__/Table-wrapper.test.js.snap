// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableActionCell component tests should render properly component output 1`] = `
<div>
  <div
    className="row justify-content-center mt-5"
  >
    <h2>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-dumbbell fa-w-20 "
        data-icon="dumbbell"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M104 96H56c-13.3 0-24 10.7-24 24v104H8c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h24v104c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm528 128h-24V120c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v272c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h24c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM456 32h-48c-13.3 0-24 10.7-24 24v168H256V56c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v400c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h128v168c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       
      test
    </h2>
  </div>
  <div>
    <ReactTable
      columns={
        Array [
          Object {},
          Object {},
          Object {
            "Cell": [Function],
            "Header": "Akcje",
            "accessor": "id",
          },
        ]
      }
      data={
        Array [
          1,
          2,
          3,
        ]
      }
    />
  </div>
</div>
`;
